app/controllers/api/v1x0/icons_controller.rb

```diff
@@ -0,0 +1,25 @@
+module Api
+  module V1x0
+    class IconsController < ApplicationController
+      include Api::V1x0::Mixins::IndexMixin
+
+      def show
+        portfolio_item = PortfolioItem.find_by!(:id => params.require(:portfolio_item_id))
+        raise ActiveRecord::RecordNotFound, "Icon not present on Portfolio Item" if portfolio_item.service_offering_icon_ref.nil?
+        send_icon_data(portfolio_item.service_offering_icon_ref)
+      rescue ActiveRecord::RecordNotFound, Catalog::TopologyError => e
+        render :json => { :message => e.message }, :status => :not_found
+      end
+
+      private
+
+      def send_icon_data(id)
+        so = ServiceOffering::Icons.new(id)
+        icon = so.process.icon
```

Comment:
Is there a way that the icon method returns the `send_data` payload? Then line 9 above could be `so.process.icon`, though I don't mind this refactor, it's just that with the private method the controller is a bit fatter than I would like. Possibly moving more logic into the Service class seems like an option.

Reply:
`send_data` actually [just sets the appropriate headers and calls `render`](https://api.rubyonrails.org/v5.2.2/classes/ActionController/DataStreaming.html#method-i-send_data) 

Would we rather just do that manually? I thought about moving the `send_icon_data` guts just back into the show method, that way we would get rid of the private method. Thats the way they're doing it on topology as well. Let me know your thoughts